src/mainloop.c

```diff
@@ -69,12 +69,48 @@ static int sock_set_non_blocking(int fd)
     return 0;
 }
 
-/* TODO: use command line options to specify */
-#define PORT 8081
 #define WEBROOT "./www"
+#define DEFAULT_PORT_NUMBER 8081
+
+static int get_port_number(char *arg_port)
+{
+    int port_number = strtol(arg_port, &arg_port, 10);
```

Comment:
Check the return values as stated in [strtol(3)](https://man7.org/linux/man-pages/man3/strtol.3.html).
You have to validate the port numbers in advance.